### PR TITLE
Add validation to `ray.ray_constants.env_integer`

### PR DESCRIPTION
**Pull request by sumanthratna**
## Why are these changes needed?

there are two copies of `env_integer`

## Related issue number

n/a

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/ray_constants.py [hunk 1]
```diff
@@ -9,7 +9,11 @@
 
 def env_integer(key, default):
     if key in os.environ:
-        return int(os.environ[key])
+        value = os.environ[key]
+        if value.isdigit():
+            return int(os.environ[key])
+        raise ValueError(f"Found {key} in environment, but value must "
+                         f"be an integer. Got: {value}.")
```
Comment (richardliaw) [inline python/ray/ray_constants.py:16]: Can you change this to just log a warning and return the default?

**Comment by sumanthratna** [comment]
@richardliaw bump—is the last commit what you expected?

**Comment by richardliaw** [comment]
hmm could you make it a debug log statement? 

```
logger.debug
```

Just worried that it could create spam.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
nice work!

**Pull request merged**